### DISCUSSION TITLE
HugeGraph Code Style

### DISCUSSION THREAD

**Discussion by imbajin** [apache/incubator-hugegraph#1848] [Polls]
After the HugeGraph join the `Apache Foundation`, as the contributor of participants increases, we need to unify the **code style** as much as possible and check the judgment of plug-ins in maven

Try to maintain a balance between  **ease of use** & Beauty & **compatibility**, which involves many controversial points, which need to be discussed together - vote

---

当下项目进入 apache 后, 随着参与人数增加, 我们需要尽可能统一代码风格, 以及检查插件的判断.

尽可能在**易用性** & 美观 & **兼容性** 中保持一个平衡, 这涉及到不少有争议的点, 需要大家一起探讨 - vote

**Comment by imbajin** [comment]
@z7658329  @seagle-yuan  @zyxxoo @mayelena @JackyYangPassion  @Fizell 

we could comment here

**Reply by chuanxo** [reply]
hard to decide, maybe we should refer to other projects and keep the same for some controversial styles

**Comment by z7658329** [comment]
Apache ShardingSphere:
https://mp.weixin.qq.com/s/mARpEtxyM9RuDgrFNsWbTA

**Reply by imbajin** [reply]
we could refer it to improve & decide our style

**Comment by javeme** [comment]
We need to keep as compatible as possible with our previous code style guidelines:
https://github.com/apache/incubator-hugegraph-doc/wiki/HugeGraph%E4%BB%A3%E7%A0%81%E9%A3%8E%E6%A0%BC%E6%8C%87%E5%8D%97

**Comment by seagle-yuan** [comment]
I udpated the checkstyle version. from v8.29(default version) to v8.45.
this version fixed lambda indentation issue

**Comment by seagle-yuan** [comment]
we shoud discuss whether  RedundantModifier OR  InterfaceMemberImpliedModifier check is needed.

https://checkstyle.sourceforge.io/config_modifier.html#InterfaceMemberImpliedModifier
https://checkstyle.sourceforge.io/config_modifier.html#RedundantModifier
